### ISSUE TITLE
[Clang] Missing diagnostics for attributes after namespace name

### ISSUE THREAD

**Issue by MagentaTreehouse** [llvm/llvm-project#121407]
The location of the attribute is non-standard in this code:
```c++
namespace N [[deprecated]] {}
```
But clang doesn't emit any diagnostics with `-Wall -Wextra -Wpedantic`.
Compiler Explorer: https://compiler-explorer.com/z/6zcPanj5Y

**Comment by shafik** [comment]
CC @erichkeane

**Comment by safocl** [comment]
```
named-namespace-definition:
    inline(opt) namespace attribute-specifier-seq(opt) identifier { namespace-body }
```
([here](https://timsong-cpp.github.io/cppwp/n4950/basic.namespace#nt:named-namespace-definition))

**Comment by a-tarasyuk** [comment]
The commit [8bd0055](https://github.com/llvm/llvm-project/commit/8bd0055) added handling attributes after the namespace identifier
